watsononlinestore/watson_online_store.py

```diff
@@ -180,6 +192,76 @@ def setup_conversation_workspace(conversation_client, environ):
         return ret
 
     @staticmethod
+    def setup_discovery_collection(discovery_client,
+                                   environment_id,
+                                   collection_id,
+                                   data_source):
+        """ Ensure that the collection exists in the Watson Discovery service.
+
+        :param discovery_client: discovery service client
+        :param str environment_id: discovery service environment id
+        :param str collection_id: discovery service collection id
+        :param str data_source: name of the discovery data source
+        :return: ID of discovery collection to use
+        :rtype: str
+        :raise Exception: When collection is not found and cannot be created
+        """
+
+        # Ensure we a valid environment exists.
+        environment = discovery_client.get_environment(environment_id)
+        if not environment:
+            raise Exception("DISCOVERY_ENVIRONMENT_ID=%s is specified in "
+                            "a runtime environment variable, but that "
+                            "environment does not exist." % environment_id)
+
+        # Determine if collection exists.
+        collection = discovery_client.get_collection(environment_id,
+                                                     collection_id)
+
+        if collection:
+            return collection_id
+
+        # Try to find collection by name.
+        for coll in \
+                discovery_client.list_collections(
+                    environment_id).get('collections', []):
+            if ((data_source == "AMAZON" and
```

Comment:
i'd prefer string checks be done like:

  AMAZON  = 'amazon'
  IBM = 'ibm_store'

  if data_source.lower() == self.AMAZON

or

  if data_source.lower() == self.IBM